pkg/innerring/innerring.go

```diff
@@ -1145,3 +1163,30 @@ func (s *Server) newEpochTickHandlers() []newEpochHandler {
 
 	return newEpochHandlers
 }
+
+func (s *Server) restartMorph() error {
```

Comment:
I've been expecting some `innerring.New()` instead of this, but looking at the way connections are managed now maybe it's not realistic...

Reply:
so that is why issue fixing is so slow. i can change it but i dont know what way is the best (for now), so after trying some ways i came up with the shortest one (111 lines only and it works)

> I've been expecting some innerring.New()

do you mean the full [New](https://github.com/notimetoname/neofs-node/blob/92c391dda95e151bc96ca55d9242b20c8406d6f4/pkg/innerring/innerring.go#L329) every time a connections is lost?

also, left https://github.com/nspcc-dev/neofs-node/issues/2410

Reply:
The problem is that we have dependencies inverted here. There can't be any IR unless there is a chain (well, two chains) and there is a connection to it. But we have this IR struct that pretends to be independent of its own state (that is inherently tied to chains) and handles connections internally. It can accept current solution if refactoring it now requires too much effort.

Reply:
> It can accept current solution if refactoring it now requires too much effort.

refactor will be done [separately](https://github.com/nspcc-dev/neofs-node/issues/2410#issuecomment-1613554057)